dask/dataframe/io/parquet/arrow.py

```diff
@@ -750,14 +750,14 @@ def initialize_write(
                 # Original dataset does not exist - cannot append
                 append = False
         if append:
-            names = dataset.metadata.schema.names
+            arrow_schema = dataset.schema.to_arrow_schema()
```

Comment:
Unrelated to this PR, but just noticed it while looking at the diff: we should try to avoid using `ParquetDataset` in the ArrowDatasetEngine (I think it's only used here in case of `append=True` to get the metadata?)

Reply:
Right, thanks for making a note of this.  This code is actually shared by all Arrow-based engines, because it is used within `to_parquet` (and my understanding is that the pyarrow.dataset API is only for reading).  It is certainly possible for us to avoid `ParquetDataset` altogether here (when the engine is "pyarrow-dataset").

On a related note - Are there plans to deprecate the `ParquetDataset` API in the near or distant future?  I just realized/remembered yesterday that both Dask-cuDF and NVTabular are still using it directly.

Reply:
`pyarrow.dataset` can also be used for writing nowadays. But, ParquetDataset is not actually used to write here, I think, but to read the metadata of the existing dataset when appending.

There are plans to deprecate ParquetDataset, yes, but for now without concrete timeline. Personally, I would first like to have dask switch the default before we actively start deprecating it (of course, we could also start deprecating it to nudge dask a bit .. ;))

Reply:
> pyarrow.dataset can also be used for writing nowadays.

Nice - Is it mature enough for us to expand the "pyarrow-dataest" engine to use it?

> But, ParquetDataset is not actually used to write here, I think, but to read the metadata of the existing dataset when appending.

Thats right - ParquetDataset is not actually used to write anything here.  My point was just that both "pyarrow-legacy" and "pyarrow-dataset" share this same `initialize_write` function when `to_parquet` is called.

> There are plans to deprecate ParquetDataset, yes, but for now without concrete timeline. Personally, I would first like to have dask switch the default before we actively start deprecating it (of course, we could also start deprecating it to nudge dask a bit .. ;))

Okay, taht makes sense.  I will start prioritizing the move to a "pyarrow" == "pyarrow-dataset" (and also start moving dask_cudf/NVTabular to the dataset backend)

Reply:
>pyarrow.dataset can also be used for writing nowadays.

Do you suggest any documentation on this? Is it still possible to extract, aggregate and write parquet filemetdata objects?  The reason we are using `ParquetDataset` is that we need the inital metadata object that we will be adding to during the append.

Reply:
Note that I opened #7544 to seperately deal with the `ParquetDataset` issue (to keep this PR/fix as simple as possible).